src/EditorFeatures/Test2/Utilities/MockDocumentNavigationServiceProvider.vb

```diff
@@ -22,33 +22,68 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.CallHierarchy
         Friend Class MockDocumentNavigationService
             Implements IDocumentNavigationService
 
-            Public DocumentId As DocumentId
-            Public TextSpan As TextSpan
+            Public ProvidedDocumentId As DocumentId
+            Public ProvidedTextSpan As TextSpan
+            Public ProvidedLineNumber As Integer
+            Public ProvidedOffset As Integer
+            Public ProvidedPosition As Integer
+            Public ProvidedVirtualSpace As Integer
+            Public ProvidedUsePreviewTab As Boolean
+
+            Public CanNavigateToLineAndOffsetReturnValue As Boolean = True
+            Public CanNavigateToPositionReturnValue As Boolean = True
+            Public CanNavigateToSpanReturnValue As Boolean = True
+
+            Public TryNavigateToLineAndOffsetReturnValue As Boolean = True
+            Public TryNavigateToPositionReturnValue As Boolean = True
+            Public TryNavigateToSpanReturnValue As Boolean = True
```

Comment:
Moq provides the ability to specify return values for mocked functions...why aren't we just using that?